### PR TITLE
Form Class: MultiSelect

### PR DESCRIPTION
**Pull request by raynichc**
## New Feature:
Form Element for selecting multiple options.

## Description
This is a form element comprising of two selects, a source and destination respectively. Options can be transferred from one to the other with buttons and optionally the lists can be sorted based on specified attributes.

## Motivation and Context
Selecting multiple options using a standard select element could be difficult and frustrating at times. This form element was design to fix this problem.

## How Has This Been Tested?
This was tested in an isolated test page. The element was tested to ensure it and all its features were functional. It was also tested to ensure that multiple multi selectors could function in the same form, as well as being tested for compatibility with all other form elements.

### CODE REVIEW

**Comment by SKuipers** [comment]
😃

File: src/Forms/Input/MultiSelect.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    protected $sortBySelect;
+
+    public function __construct(FormFactoryInterface &$factory, $name) {        
+        $this->name = $name;
+
+        $this->sourceSelect = $factory->createSelect($name . "Source")->selectMultiple(true)->setSize(8)->setClass('mediumWidth')->addClass("floatNone");
+        $this->destinationSelect = $factory->createSelect($name . "Destination")->selectMultiple(true)->setSize(8)->setClass('mediumWidth');
+
+        $this->sortBySelect = $factory->createSelect($name . "Sort")
+            ->placeholder("Sort by Name")
```
Comment (SKuipers) [inline src/Forms/Input/MultiSelect.php]: String needs wrapped in __() for translation

File: src/Forms/Input/MultiSelect.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            ->setClass("shortWidth");
+        $this->removeButton = $factory->createButton("Remove")
+            ->onClick('optionTransfer(\'' . $this->destinationSelect->getID() . '\',\'' . $this->sourceSelect->getID() . '\')')
+            ->setClass("shortWidth");
+    }
+
+    public function addSortableAttribute($attribute, $values)
+    {
+        $this->sortableAttributes[$attribute] = $values;
+        $this->sortBySelect->fromArray(array($attribute => "Sort by " . $attribute));
```
Comment (SKuipers) [inline src/Forms/Input/MultiSelect.php]: The "Sort By" here too. We missed a few __() 😄

File: src/Forms/Input/MultiSelect.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+        $this->sourceSelect = $factory->createSelect($name . "Source")->selectMultiple(true)->setSize(8)->setClass('mediumWidth')->addClass("floatNone");
+        $this->destinationSelect = $factory->createSelect($name . "Destination")->selectMultiple(true)->setSize(8)->setClass('mediumWidth');
+
+        $this->sortBySelect = $factory->createSelect($name . "Sort")
+            ->placeholder("Sort by Name")
+            ->setClass("shortWidth")
+            ->addClass("floatNone");
+
+        $this->addButton = $factory->createButton("Add")
```
Comment (SKuipers) [inline src/Forms/Input/MultiSelect.php]: Another __() string

File: src/Forms/Input/MultiSelect.php [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            ->onClick('optionTransfer(\'' . $this->sourceSelect->getID() . '\',\'' . $this->destinationSelect->getID() . '\')')
+            ->setClass("shortWidth");
+        $this->removeButton = $factory->createButton("Remove")
```
Comment (SKuipers) [inline src/Forms/Input/MultiSelect.php]: Another one for __()

**Comment by rossdotparker** [comment]
@SKuipers it looks like you've given this a once over, which is great. There are conflicts due to other PRs merged out of turn. Are you OK to resolve the conflicts and then merge? I'd rather you do it, as this section of the codebase is most definitely yours. Thanks!

**Comment by raynichc** [comment]
No problem. Just pushed it, merged and resolved.

**Pull request merged**